src/MSBuild.UnitTests/XMake_Tests.cs

```diff
@@ -809,6 +809,41 @@ public void BuildFailsWithBadPropertyName()
             results.ShouldContain("error MSB4177");
         }
 
+        [Theory]
+        [InlineData("-getProperty:Foo", "propertyContent")]
+        [InlineData("-getItem:Bar", "ItemContent")]
+        [InlineData("-getTargetResult:Biz", "Success")]
+        public void GetStarOutputsToFileIfRequested(string extraSwitch, string result)
+        {
+            using TestEnvironment env = TestEnvironment.Create();
+            TransientTestFile project = env.CreateFile("testProject.csproj", @"
+<Project>
+  <PropertyGroup>
+    <Foo>propertyContent</Foo>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Bar Include=""ItemContent"" />
+  </ItemGroup>
+
+  <Target Name=""Biz"" />
+</Project>
+");
+            string resultFile = Path.Combine(Path.GetDirectoryName(project.Path), "resultFile.txt");
```

Comment:
```suggestion
            string resultFile = env.GetTempFile(".txt").Path;
```

will delete the file automatically when the test environment is disposed of, so no need for the `try`/`finally` below.